include/SFML/Graphics/Color.hpp

```diff
@@ -29,6 +29,7 @@
 // Headers
 ////////////////////////////////////////////////////////////
 #include <SFML/Graphics/Export.hpp>
+#include <algorithm>
```

Comment:
Exposing this mass-header is a bit unfortunate for compile times. But as you suggested [in the other issue](https://github.com/SFML/SFML/issues/1893#issuecomment-990180997), for `min`/`max` alone we could have our own lightweight implementation, in a later PR.

Reply:
You are a fast reviewer :) I recognized this issue and changed my approach, implementing some local lambdas to deal with the casting and clamping of the various operations. I think the end result is cleaner than before as we avoid quite a lot of C-style casting, and we completely avoid the dependency on `<algorithm>`, which is indeed a compile-time hog.